src/MoBi.Presentation/MenusAndBars/ContextMenus/ContextMenuFactoryForNeighborhoods.cs

```diff
@@ -0,0 +1,57 @@
+﻿using MoBi.Core.Domain.Model;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Presenter;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Presentation.Core;
+using OSPSuite.Presentation.MenuAndBars;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Presentation.Presenters.ContextMenus;
+using OSPSuite.Presentation.Repositories;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Presentation.MenusAndBars.ContextMenus
+{
+   public class ContextMenuFactoryForNeighborhoods : ContextMenuForObjectBaseDTOSpecificationFactory<IContainer>
+   {
+      public ContextMenuFactoryForNeighborhoods(IMoBiContext context) : base(context)
+      {
+      }
+
+      public override IContextMenu CreateFor(ObjectBaseDTO objectBaseDTO, IPresenterWithContextMenu<IViewItem> presenter)
+      {
+         var contextMenu = _context.Resolve<ContextMenuForNeighborhoods>();
+         return contextMenu.InitializeWith(objectBaseDTO, presenter);
+      }
+
+      protected override bool IsSatisfiedBy(IEntity entity, IPresenterWithContextMenu<IViewItem> presenter)
+      {
+         return presenter.IsAnImplementationOf<IHierarchicalSpatialStructurePresenter>()
```

Comment:
don;t really lile this, Maybe we can just use the fact that it;s a container named neigthborhood